### ISSUE TITLE
Sharpen: no known route from 'labs' to 'matrix'

### ISSUE THREAD

**Issue by maximeg** [libvips/libvips#1428]
Hi,

I just upgraded from 8.8.1 to 8.8.2 and my specs are failing with the following vips error:

```
       vips_realpath: unable to form filename
       unix error: No such file or directory
       vips_colourspace: no known route from 'labs' to 'matrix'
```

I test with several images and for all pipelines including a sharpen, this fails.

The sharpen:

```ruby
       vips_image.sharpen(
         sigma: 1,
         x1: 2,
         y2: 10,
         y3: 20,
         m1: 0,
         m2: 3
       )
```

**Comment by jcupitt** [comment]
Yes, this was a bad idea in 8.8.2, I didn't realize it would break things. Sorry @maximeg.

It's reverted in 8.8.3. I'll push out a new libvips now.

**Comment by jcupitt** [comment]
There's an 8.83 pre-release. I'll add some win binaries tonight.

https://github.com/libvips/libvips/releases/tag/v8.8.3

**Comment by maximeg** [comment]
Oh, I missed the release in preparation. Thanks for your reactivity and the release!

I just checked that everything is fine with 8.8.3 (brew distribution was quickly updated !).

Question around the error message: I was loading the images from a buffer and in the error message I mentioned, there was `vips_realpath: unable to form filename` and `unix error: No such file or directory`. Is there a file operation that should not happen (and otherwise is silent)?

**Comment by jcupitt** [comment]
Sorry, I don't know what could be calling realpath there. I tried:

```python
import sys
import pyvips

buf = open(sys.argv[1], "rb").read()
x = pyvips.Image.new_from_buffer(buf, "")
x = x.sharpen()
buf2 = x.write_to_buffer(".jpg")
```

Then:

```
john@yingna ~/try $ gdb python3
Reading symbols from python3...
(No debugging symbols found in python3)
(gdb) break vips_realpath
Function "vips_realpath" not defined.
Make breakpoint pending on future shared library load? (y or [n]) y
Breakpoint 1 (vips_realpath) pending.
(gdb) run sharpen.py ~/pics/k2.jpg
Starting program: /usr/bin/python3 sharpen.py ~/pics/k2.jpg
(gdb) 
```

ie. `vips_realpath` is not called by anything obvious here. We'd need a test-case to debug that further.

**Comment by jcupitt** [comment]
(closed since 8.8.3 is now now with the fix)
